botocore/auth.py

```diff
@@ -241,7 +241,10 @@ def payload(self, request):
             request.body.seek(position)
             return hex_checksum
         elif request.body:
-            return sha256(request.body.encode('utf-8')).hexdigest()
+            body = request.body
```

Comment:
What about the first `if` condition?  Looks like it only supports bytes.  Do we need to worry about that case as well?